script.js

```diff
@@ -1,4 +1,118 @@
+// Base & comfortable only
+
+//global variables
+var gameMode = "inputUsername";
+var player1Name = "";
+var player2Name = "";
+var user1Choice = 0;
+var user2Choice = 0;
+var userDice1 = 0;
+var userDice2 = 0;
+
+// Greet user before games starts
+var greetUser = function (userNameInput) {
+  greetMessage = `Hi ${userNameInput}, welcome to a game of Beat That! <br> Please click the submit Button to roll your dices`;
+  return greetMessage;
+};
+
+// Let User know Dices & let them choose
+var userChooseSequence = function (dice1, dice2) {
+  string1 = dice1.toString();
+  string2 = dice2.toString();
+  choice1 = `${string1}${string2}`;
+  choice2 = `${string2}${string1}`;
+  message = `You have rolled <br>First Dice: ${dice1} <br>Second Dice: ${dice2} <br> Please type in your choice of either '${choice1}' or '${choice2}' and click submit to lock in your choice!`;
+  return message;
+};
+
+// Roll Dice Function
+var rollDice = function () {
+  var randomInteger = Math.floor(Math.random() * 6);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+// Compare Player 1 & 2 choices and output
+var tabulateResults = function (
+  player1Name,
+  player2Name,
+  player1score,
+  player2score
+) {
+  if (player1score > player2score) {
+    results = `${player1Name}'s score is ${player1score} <br>${player2Name}'s score is ${player2score} <br> ${player1Name} wins! Congratulations! <br> Please click Submit to play again!`;
+  } else {
+    results = `${player1Name}'s score is ${player1score} <br>${player2Name}'s score is ${player2score} <br> ${player2Name} wins! Congratulations! <br> Please click Submit to play again!`;
+  }
+
+  return results;
+};
+
+// // input validation
+// var inputValidation = function (input, dice1, dice2) {
+//   string1 = dice1.toString();
+//   string2 = dice2.toString();
+//   choice1 = `${string1}${string2}`;
+//   choice2 = `${string2}${string1}`;
+//   if (input == choice1 || input == choice2) {
+//     input = Number(input);
+//   } else {
+//     myOutputValue = `ERROR! Only type in the choices given man`;
+//     console.log(myOutputValue);
+//     return myOutputValue;
```

Comment:
Do note that the return of myOutputValue here (though necessary), will not end the code sequence in the main function since this is only within the helper function. Should you attempt to implement this input validation function, consider having another gamemode (e.g., "invalid input") so that the main function can break out of the code sequence. Do remember to change the gamemode back to "lockInChoice" after that so that the user comes back to input their choice again.